cranelift-codegen/meta/src/shared/instructions.rs

```diff
@@ -1159,6 +1159,27 @@ pub fn define(
         .can_load(true),
     );
 
+    ig.push(
+        Inst::new(
+            "fill_nop",
+            r#"
+        This is identical to :inst:`fill`, except it has no encoding, since
+        it is a no-op.
+
+        This instruction is created only during late-stage peephole
+        optimisation, after all registers and stack slots have been assigned.
```

Comment:
As discussed, suggest "reload optimisation" instead of "peephole optimisation" in order not to muddy the waters more than necessary.